### PR TITLE
Support security plugin enabled testing in CI

### PR DESCRIPTION
**Pull request by kaituo**
*Issue #, if available:*
https://github.com/opendistro-for-elasticsearch/anomaly-detection/issues/142

*Description of changes:*
This change is to support integration testing against remote security-enabled clustering in our CI. Currently, we only support the integration testing without a security plugin and have to remove the security plugin in CI.

The critical change is to create an https client with basic auth in ODFERestTestCase and let our rest test case inherit from ODFERestTestCase. With default security configuration, the password for the admin user is also “admin.” For ssl certification, we choose to disable it since it’s the default demo cert used here.

Testing done:
* ran the scripts on CI workflow on Mac and tested it works.
* gradle build still passes.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -63,9 +62,16 @@ jobs:
...
           docker run -p 9200:9200 -d -p 9600:9600 -e "discovery.type=single-node" odfe-ad:test
           sleep 90
-          curl -XGET http://localhost:9200/_cat/plugins
 
       - name: Run AD Test
         if: env.imagePresent == 'true'
         run: |
-          ./gradlew :integTestRunner --tests "com.amazon.opendistroforelasticsearch.ad.rest.*IT" --tests "com.amazon.opendistroforelasticsearch.ad.e2e.*IT" -Dtests.rest.cluster=localhost:9200 -Dtests.cluster=localhost:9200 -Dtests.clustername="docker-cluster"
+          security=`curl -XGET https://localhost:9200/_cat/plugins?v -u admin:admin --insecure |grep opendistro_security|wc -l`
+          if [ $security -gt 0 ]
```
Comment (weicongs-amazon) [inline .github/workflows/CI.yml:70]: we can consider having both IT with security and without security running. This is the way we are using the ODFE building workflow. The team can decide which one is better

File: src/test/java/com/amazon/opendistroforelasticsearch/ad/ODFERestTestCase.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        }
+
+        builder.setStrictDeprecationMode(true);
+        return builder.build();
+    }
+
+    @SuppressWarnings("unchecked")
+    @After
+    protected void wipeAllODFEIndices() throws IOException {
+        Response response = client().performRequest(new Request("GET", "/_cat/indices?format=json"));
```
Comment (weicongs-amazon) [inline src/test/java/com/amazon/opendistroforelasticsearch/ad/ODFERestTestCase.java:86]: not sure if AD plugin is using any hidden index or not. I got this issue for alerting plugin. https://github.com/opendistro-for-elasticsearch/alerting/pull/213/files#diff-e1e03fb0ddd2ab303d2631b957c116e1R64

Comment (kaituo) [reply src/test/java/com/amazon/opendistroforelasticsearch/ad/ODFERestTestCase.java:86]: added

File: .github/workflows/CI.yml (hunk 1 above)
Comment (kaituo) [reply .github/workflows/CI.yml:70]: As discussed offline, gradle build already runs IT without security enabled.  Will keep CI as it is for now.

File: src/test/java/com/amazon/opendistroforelasticsearch/ad/ODFERestTestCase.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+ * ODFE integration test base class to support both security disabled and enabled ODFE cluster.
+ */
+public abstract class ODFERestTestCase extends ESRestTestCase {
+
+    protected boolean isHttps() {
+        boolean isHttps = Optional.ofNullable(System.getProperty("https")).map("true"::equalsIgnoreCase).orElse(false);
+        if (isHttps) {
+            // currently only external cluster is supported for security enabled testing
+            if (!Optional.ofNullable(System.getProperty("tests.rest.cluster")).isPresent()) {
+                throw new RuntimeException("external cluster url should be provided for security enabled testing");
```
Comment (yizheliu-amazon) [inline src/test/java/com/amazon/opendistroforelasticsearch/ad/ODFERestTestCase.java:58]: `external` might be confusing as in the CI we are actually using `localhost`. Maybe we can just say "cluster url should be provided for security enabled testing"

Comment (kaituo) [reply src/test/java/com/amazon/opendistroforelasticsearch/ad/ODFERestTestCase.java]: make sense.  changed.

**Pull request merged**